pkg/operator/operator.go

```diff
@@ -158,7 +158,9 @@ func (optr *Operator) handleErr(err error, key interface{}) {
 		optr.queue.AddRateLimited(key)
 		return
 	}
-
+	if err := optr.statusFailing(err.Error()); err != nil {
```

Comment:
I don't grok this: we call `statusFailing()` and log an error if this fails. And then, given the scoping rules, we continue to use/consume `err` as passed in. Where do we "set operator status failing if sync operator fails"?